rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ThreadDumpView.java

```diff
@@ -75,144 +72,98 @@ public ThreadDumpView(RMController controller) {
-        this.noNodeLabel.show();
-        this.layout.hide();
         this.nodeUrl = null;
         this.nodeSourceName = null;
         this.nodeHostName = null;
+
+        this.nodeLabel.setContents(NO_NODE_SELECTED_MESSAGE);
+        this.nodeLabel.setIcon("");
+        this.nodeThreadDumpButton.disable();
+        resetThreadDumpArea();
     }
 
+    @Override
     public void nodeSelected(NodeSource.Host.Node node) {
-        this.nodeLabel.setIcon(node.getIcon());
-
-        this.noNodeLabel.hide();
-
         this.nodeUrl = node.getNodeUrl();
         this.nodeSourceName = node.getSourceName();
         this.nodeHostName = node.getHostName();
+
         this.nodeLabel.setContents("<h3>" + node.getNodeUrl() + "</h3>");
+        this.nodeLabel.setIcon(node.getIcon());
+        this.nodeThreadDumpButton.enable();
         resetThreadDumpArea();
-
-        this.layout.show();
     }
 
+    @Override
     public void nodeSourceSelected(NodeSource ns) {
         this.nodeUrl = null;
-        this.layout.hide();
-        this.noNodeLabel.show();
+        this.nodeHostName = null;
+
+        this.nodeLabel.setContents(NO_NODE_SELECTED_MESSAGE);
+        this.nodeLabel.setIcon("");
+        this.nodeThreadDumpButton.disable();
+        resetThreadDumpArea();
     }
 
+    @Override
     public void hostSelected(NodeSource.Host h) {
         this.nodeUrl = null;
-        this.layout.hide();
-        this.noNodeLabel.show();
+        this.nodeSourceName = null;
+
+        this.nodeLabel.setContents(NO_NODE_SELECTED_MESSAGE);
+        this.nodeLabel.setIcon("");
+        this.nodeThreadDumpButton.disable();
+        resetThreadDumpArea();
     }
 
     @Override
     public void nodesUpdated(Map<String, NodeSource> nodes) {
         NodeLabel.update(nodes, this.nodeLabel, this.nodeSourceName, this.nodeHostName, this.nodeUrl);
     }
 
-    private void createLayout(HLayout executeAndLoading) {
-        this.layout = new VLayout(7);
-        this.layout.setWidth100();
-        this.layout.setHeight100();
-
-        this.layout.addMember(this.nodeLabel);
-        this.layout.addMember(executeAndLoading);
-        this.layout.addMember(this.threadDumpArea);
+    private void initLabels() {
+        this.nodeLabel = new Label();
+        this.nodeLabel.setHeight(25);
+        this.nodeLabel.setContents(NO_NODE_SELECTED_MESSAGE);
 
-        this.layout.hide();
+        this.loadingLabel = new Label();
+        this.loadingLabel.setHeight(25);
+        this.loadingLabel.setIcon("loading.gif");
+        this.loadingLabel.hide();
     }
 
-    private void createThreadDumpButtons() {
+    private void initThreadDumpButtons() {
         this.nodeThreadDumpButton = new IButton("Fetch Node's Thread Dump");
-        this.nodeThreadDumpButton.setWidth("220px");
+        this.nodeThreadDumpButton.setWidth("160px");
         this.nodeThreadDumpButton.addClickHandler(event -> {
             if (this.nodeUrl != null) {
                 String nodeUrl = this.nodeUrl;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "nodeUrl" which hides the field declared at line 59. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)